pandas/src/inference.pyx

```diff
@@ -686,9 +686,9 @@ def maybe_convert_numeric(object[:] values, set na_values,
                             raise ValueError('integer out of range')
                     else:
                         seen_float = True
-            except:
+            except (TypeError, ValueError) as e:
                 if not coerce_numeric:
-                    raise
+                    raise type(e)(e.message + ' in row {}'.format(i))
```

Comment:
acutally maybe row is not the word, these are 1-d arrays, so maybe entry? value at index i?